command/quota_apply.go

```diff
@@ -296,9 +299,41 @@ func parseQuotaResource(result *api.Resources, list *ast.ObjectList) error {
 		}
 	}
 
+	// Parse storage block
+	storageBlocks := listVal.Filter("storage")
+	storage, err := parseStorageResource(storageBlocks)
+	if err != nil {
+		return multierror.Prefix(err, "storage ->")
+	}
+	result.Storage = storage
+
 	return nil
 }
 
+func parseStorageResource(storageBlocks *ast.ObjectList) (*api.QuotaStorageResources, error) {
+	switch len(storageBlocks.Items) {
+	case 0:
+		return nil, nil
+	case 1:
+	default:
+		return nil, errors.New("only one storage block is allowed")
+	}
```

Comment:
this still feels funny when we could avoid the concern, and the extra parsing, by using a map instead of a block, but I suppose consistency with the surrounding context wins out.